src/kernel.ts

```diff
@@ -1,1069 +1,784 @@
-        // to depend on the Notebook or Pager classes.
-        for (var i=0; i<l; i++) {
-            var payload = payloads[i];
-            var callback = payload_callbacks[payload.source];
-            if (callback) {
-                promise.push(callback(payload, msg));
-            }
-        }
-        return Promise.all(promise);
-    }
-
-    /**
-     * @function _handle_status_message
-     */
-    _handle_status_message(msg): void {
-        var execution_state = msg.content.execution_state;
-        var parent_id = msg.parent_header.msg_id;
-        
-        // dispatch status msg callbacks, if any
-        var callbacks = this.get_callbacks_for_msg(parent_id);
-        if (callbacks && callbacks.iopub && callbacks.iopub.status) {
-            try {
-                callbacks.iopub.status(msg);
-            } catch (e) {
-                console.log("Exception in status msg handler", e, e.stack);
-            }
-        }
-        
-        if (execution_state === 'busy') {
-            events.trigger('kernel_busy.Kernel', {kernel: this});
-
-        } else if (execution_state === 'idle') {
-            // signal that iopub callbacks are (probably) done
-            // async output may still arrive,
-            // but only for the most recent request
-            this._finish_iopub(parent_id);
-            
-            // trigger status_idle event
-            events.trigger('kernel_idle.Kernel', {kernel: this});
-
-        } else if (execution_state === 'starting') {
-            events.trigger('kernel_starting.Kernel', {kernel: this});
-            this.kernel_info((reply) => {
-                this.info_reply = reply.content;
-                events.trigger('kernel_ready.Kernel', {kernel: this});
-            });
-
-        } else if (execution_state === 'restarting') {
-            // autorestarting is distinct from restarting,
-            // in that it means the kernel died and the server is restarting it.
-            // kernel_restarting sets the notification widget,
-            // autorestart shows the more prominent dialog.
-            this._autorestart_attempt = this._autorestart_attempt + 1;
-            events.trigger('kernel_restarting.Kernel', {kernel: this});
-            events.trigger('kernel_autorestarting.Kernel', {kernel: this, attempt: this._autorestart_attempt});
-
-        } else if (execution_state === 'dead') {
-            events.trigger('kernel_dead.Kernel', {kernel: this});
-            this._kernel_dead();
-        }
-    }
-    
-    /**
-     * Handle clear_output message
-     *
-     * @function _handle_clear_output
-     */
-    _handle_clear_output(msg): void {
-        var callbacks = this.get_callbacks_for_msg(msg.parent_header.msg_id);
-        if (!callbacks || !callbacks.iopub) {
-            return;
-        }
-        var callback = callbacks.iopub.clear_output;
-        if (callback) {
-            callback(msg);
-        }
-    }
+      }
+    };
```

Comment:
No, switching tabs, doing other things, and going to move to BIDS. DOE library open only at 9 in summer, so stuck in Coffee shop for now, but doors should open any time soon. :-)